mesonbuild/modules/windows.py

```diff
@@ -62,7 +68,22 @@ def _find_resource_compiler(self, state):
         if not rescomp.found():
             raise MesonException('Could not find Windows resource compiler')
 
-        self._rescomp = rescomp
+        rescomp_type = None
+        for (arg, match, type) in [
+                ('/?', '^.*Microsoft.*Resource Compiler.*$', ResourceCompilerType.rc),
+                ('--version', '^GNU windres.*$', ResourceCompilerType.windres),
```

Comment:
I'd prefer if you could drop the `^` here. In LLVM-MinGW, I have a windres-compatible shellscript wrapper around `llvm-rc`, and this frontend does print `GNU windres`, but not directly at the beginning of a line:
https://github.com/mstorsjo/llvm-mingw/blob/master/wrappers/windres-wrapper.sh#L44

Reply:
Ah, good catch. Well, I can change the `--version` output as well (I just had to amend it the other day to add support for `--preprocessor-arg=`). I could make the `--version` output be e.g. `LLVM windres (GNU windres compatible)`.